### DISCUSSION TITLE
TextField.text=number 0 does not shown in iOS.

### DISCUSSION THREAD

**Discussion by ghost** [NativeScript/NativeScript#9846] [General]
https://github.com/NativeScript/NativeScript/blob/f88c158b384fee7afec9770b257a4312e715b6ec/packages/core/ui/text-base/index.ios.ts#L130
There is difference between iOS and Android.
https://github.com/NativeScript/NativeScript/blob/f88c158b384fee7afec9770b257a4312e715b6ec/packages/core/ui/text-base/index.android.ts#L259
Since number may be assigned to TextField.text (For example, ListView's item has number 0 field), it is desirable to match Android.

**Comment by lostation** [comment]
Yes indeed, I'm facing the same weird behavior on iOS... again...

**Comment by lostation** [comment]
But... I tried your changes...and it doesn't seem to do the trick... : (

**Comment by ghost** [comment]
It certainly seems that way.
I tried with a new project in NativeScript 8.2.1 and could not reproduce the problem.
In that version, even if I pass a number, it still says typeof value === object. I do not know why.

But I am still worried that it is not right.

**Comment by ghost** [comment]
@lostation
I found this fact.

```xml
<Repeater items="{{ ([42]) }}">
  <Repeater.itemTemplate>
    <Label text="{{ $value, `1st:${$value == 42} 2nd:${$value === 42}` }}"/>
  </Repeater.itemTemplate>
</Repeater>

<!--
  As an added bonus, I also found other facts `,` and `=`  like bellow causes problems.
  <Label text="{{ $value, `1st:${$value == 42}, 2nd:${$value === 42}` }}"/>
  <Label text="{{ $value, `=1st:${$value == 42} 2nd:${$value === 42}` }}"/>
-->
```

The result is unexpectedly `1st:true 2nd:false`.


The reason is that it is converted to an object type here.
https://github.com/NativeScript/NativeScript/blob/main/packages/core/ui/core/bindable/index.ts#L176-L189

I don't know how to fix it correctly, but I have worked around this problem with such a patch.
```diff
 export function convertExpressionToValue(expression, model, isBackConvert, changedModel) {
     if (!(expression.type in expressionParsers)) {
         throw Error('Invalid expression syntax');
     }
-    return expressionParsers[expression.type](expression, model, isBackConvert, changedModel);
+    const value = expressionParsers[expression.type](expression, model, isBackConvert, changedModel);
+    return (value instanceof Number || value instanceof Boolean || value instanceof String) ? value.valueOf() : value;
}
```
